components/gravityforms/class-gf-convertkit.php

```diff
@@ -211,51 +212,40 @@ public function get_custom_fields() {
 	}
 
 	/**
-	 * GF Settings callback
+	 * Return an array of ConvertKit forms
 	 *
-	 * Build a SELECT to be shown in Feed Settings to select ConvertKit form
-	 * form data will be posted to.
-	 *
-	 * @param array $field
-	 * @param bool|true $echo
-	 *
-	 * @return string
+	 * @return array List of CK Forms
 	 */
-	public function settings_convertkit_form( $field, $echo = true ) {
+	public function get_convertkit_forms() {
 		$forms = ckgf_convertkit_api_get_forms();
 
 		ckgf_debug( $forms );
 
 		if ( is_wp_error( $forms ) ) {
-			$markup = sprintf( '%s: %s', __( 'Error', 'convertkit' ), $forms->get_error_message() );
+			$error = sprintf( '%s: %s', __( 'Error', 'convertkit' ), $forms->get_error_message() );
+      ckgf_debug( $error );
+      return array();
 		} elseif ( empty( $forms ) ) {
-			$markup = sprintf( '%s: %s', __( 'Error', 'convertkit' ), __( 'Please configure some forms on ConvertKit', 'convertkit' ) );
-		} else {
-			$options = array(
-				array(
-					'label' => __( 'Select a ConvertKit form', 'convertkit' ),
-					'value' => '',
-				),
-			);
-
-			foreach ( $forms as $form ) {
-				$options[] = array(
-					'label' => esc_html( $form['name'] ),
-					'value' => esc_attr( $form['id'] ),
-				);
-			}
-
-			$markup = $this->settings_select(array_merge($field, array(
-				'choices' => $options,
-				'type'    => 'select',
-			)), false);
-		}
-
-		if ( $echo ) {
-			echo $markup;
+			$error = sprintf( '%s: %s', __( 'Error', 'convertkit' ), __( 'Please configure some forms on ConvertKit', 'convertkit' ) );
```

Comment:
Logging errors instead of them just being ephemeral is great 👍🏾 but shouldn't they also be visible to the user? Especially this.

Reply:
I agree, I’m not sure what’d be a good way from inside this function though.

Reply:
I don't know enough about Gravity Forms to say but it seems like before we were generating markup directly?

Reply:
That's correct, but the new version is only returning an array of choices, not generating the markup. We might be able to `echo` directly, but that would probably cause some issues.

Reply:
@bradpauly This is part of why I went with the fix from the patch (converting the object to an array) - it seemed worth it to keep the old UI/surfacing errors and prompts to the user